src/__tests__/environment-variables/environment-variables.test.tsx

```diff
@@ -36,52 +39,62 @@ describe('Environment variables', () => {
       );
       expectDOMElementsToMatch(actual, expected);
     });
+
+    it('Without dotenv file', async () => {
+      const { serverRender } = await getPage({
+        nextRoot: path.join(__dirname, '__fixtures__', 'no-dotenv-file'),
+        route: '/page',
+      });
+      const { nextRoot: actual } = serverRender();
+      const { container: expected } = renderWithinNextRoot(
+        <EnvVarsCleanupPage
+          envVarsMock={{
+            FROM_RUNTIME: 'FROM_RUNTIME',
+            NEXT_PUBLIC_FROM_RUNTIME: 'NEXT_PUBLIC_FROM_RUNTIME',
+            NAME_CLASH_RUNTIME_VS_CONFIG: 'FROM_RUNTIME',
+            NAME_CLASH_RUNTIME_VS_DOTFILE: 'FROM_RUNTIME',
+            NAME_CLASH_RUNTIME_VS_CONFIG_VS_DOTFILE: 'FROM_RUNTIME',
+          }}
+        />
+      );
+      expectDOMElementsToMatch(actual, expected);
+    });
   });
 
-  describe('public runtime', () => {
+  describe('client runtime', () => {
     it('Env vars from dot env file only available when prefixed with "NEXT_PUBLIC_"', async () => {
       const { render } = await getPage({
-        nextRoot: __dirname + '/__fixtures__' + '/env-vars',
+        nextRoot: path.join(__dirname, '__fixtures__', 'env-vars'),
         route: '/page',
         dotenvFile: '.env.test',
       });
       const { nextRoot: actual } = render();
       const { container: expected } = renderWithinNextRoot(
         <Page
           envVarsMock={{
-            FROM_RUNTIME: 'FROM_RUNTIME',
-            FROM_CONFIG: 'FROM_CONFIG',
-            FROM_DOTFILE: undefined,
-            FROM_DOTFILE_EXPANDED: undefined,
+            NEXT_PUBLIC_FROM_RUNTIME: 'NEXT_PUBLIC_FROM_RUNTIME',
             NEXT_PUBLIC_FROM_DOTFILE: 'NEXT_PUBLIC_FROM_DOTFILE',
-
-            NAME_CLASH_RUNTIME_VS_CONFIG: 'FROM_CONFIG',
-            NAME_CLASH_RUNTIME_VS_DOTFILE: 'FROM_RUNTIME',
-            NAME_CLASH_RUNTIME_VS_CONFIG_VS_DOTFILE: 'FROM_CONFIG',
```

Comment:
Looking at this test it seems our env loading didn't work properly before. We are doing client `render` and we got all the variables into the DOM (including those not prefixed with `NEXT_PUBLIC_`)